src/main/groovy/io/seqera/wave/redis/RedisService.groovy

```diff
@@ -0,0 +1,76 @@
+/*
+ *  Wave, containers provisioning service
+ *  Copyright (c) 2023-2024, Seqera Labs
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.seqera.wave.redis
+
+import java.util.concurrent.TimeoutException
+
+import redis.clients.jedis.Connection
+import redis.clients.jedis.StreamEntryID
+import redis.clients.jedis.Transaction
+import redis.clients.jedis.params.ScanParams
+import redis.clients.jedis.params.SetParams
+import redis.clients.jedis.params.XAutoClaimParams
+import redis.clients.jedis.params.XReadGroupParams
+import redis.clients.jedis.resps.ScanResult
+import redis.clients.jedis.resps.StreamEntry
+import redis.clients.jedis.resps.Tuple
+/**
+ * Implements RedisService
+ *
+ * @author Munish Chouhan <munish.chouhan@seqera.io>
+ */
+interface RedisService {
```

Comment:
Not sure to understand the need for this?

Reply:
JedisCluster and JedisPool doesnot have common Interface, so not possible to inject 
Thats why I have created this service which have separate implementation for cluster and standalone and based on redis.mode it will be initialized